styx-common/src/test/java/com/spotify/styx/util/ShardedCounterTest.java

```diff
@@ -226,6 +226,34 @@ public void shouldDecrementShardWithExcessUsage() {
     assertEquals(1L, shardedCounter.getCounter(COUNTER_ID1));
   }
 
+  @Test
+  public void shouldDecrementShardWithALotOfExcessUsage() {
```

Comment:
Is this an implementation of `decreaseLimitBelowCurrentCounterValue` ? Should we then remove `decreaseLimitBelowCurrentCounterValue` and the TODO?